server/server.js

```diff
@@ -1,15 +1,229 @@
+
 const express = require("express");
 const app = express();
-const port = process.env.PORT || 5000;
+const cors = require ("cors");
+app.use(cors());
+const bodyParser = require("body-parser")
+app.use(bodyParser.json())
+const videos = require("../exampleresponse.json"); 
+app.use(express.json());
+const { body, validationResult } = require("express-validator");
+const port = process.env.PORT || 3000;
+const { Pool } = require("pg");
+
+// const db = new Pool({
+//   user: "shadifakhri", 
+//   host: "localhost",
+//   database: "database",
+//   password: "",
+//   port: 5432,
+// });
+
+const db = new Pool({
+  user: "shadi_user",
+  host: "dpg-cjs7s5dv2qks738v6ltg-a.oregon-postgres.render.com",
+  database: "shadi",
+  password: "j7lLUQoGj3mMw7Q7gH66Nz3ygDizGews",
+  port: 5432,
+  ssl: {
+    rejectUnauthorized: false
+  }
+});
 
-app.listen(port, () => console.log(`Listening on port ${port}`));
 
-// Store and retrieve your videos from here
-// If you want, you can copy "exampleresponse.json" into here to have some data to work with
-let videos = [];
+// app.get("/videos", function (req, res) {
+//   if (videos.length === 0) {
+//     return res.status(404).json({ error: "no videos found" });
+//   }
+//   res.json(videos);
+// });
 
-// GET "/"
-app.get("/", (req, res) => {
-  // Delete this line after you've confirmed your server is running
-  res.send({ express: "Your Backend Service is Running" });
+// GET "/videos"
+app.get("/videos", async function (req, res) {
+  const result = await db.query("SELECT * FROM videos");
+  if (result.rows.length === 0) {
+    return res.status(404).json({ error: "no videos found" });
+  }
+  res.json(result.rows);
 });
+
+
+
+
+//ordering by assending and dessending for example /videos?order=asc
+// app.get("/videos", function (req, res) {
+//   if (videos.length === 0) {
+//     return res.status(404).json({ error: "no videos found" });
+//   }
+//   const order = req.query.order;
+
+//   let orderVideos;
+//   if (order === "asc") {
+//     orderVideos = videos.sort((a, b) => a.rating - b.rating);
+//   } else {
+//     // Default to descending order if order is not specified or is invalid
+//     orderVideos = videos.sort((a, b) => b.rating - a.rating);
+//   }
+
+//   return res.json(orderVideos);
+// });
+//ordering by assending and dessending for example /videos?order=asc
+app.get("/videos", async function (req, res) {
```

Comment:
Why do you have a second handler function for the same method on the same path? Which one will end up getting used?